pvlib/irradiance.py

```diff
@@ -3117,3 +3117,70 @@ def complete_irradiance(solar_zenith,
                                      'dhi': dhi,
                                      'dni': dni})
     return component_sum_df
+
+
+def pvl_louche(ghi, solar_zenith, datetime_or_doy):
+    """
+    Determine DNI and GHI from GHI using louche model.
+
+    Parameters
+    ----------
+    ghi : Series
+        Global horizontal irradiance. [W/m^2]
+
+    zenith : Series
+        True (not refraction-corrected) zenith angles in decimal
+        degrees. Angles must be >=0 and <=180.
+
+    datetime_or_doy : numeric, pandas.DatetimeIndex
+        Day of year or array of days of year e.g.
+        pd.DatetimeIndex.dayofyear, or pd.DatetimeIndex.
+
+    Returns
+    -------
+    data: OrderedDict or DataFrame
+        Contains the following keys/columns:
+
+            * ``dni``: the modeled direct normal irradiance in W/m^2.
+            * ``dhi``: the modeled diffuse horizontal irradiance in
+              W/m^2.
+            * ``kt``: Ratio of global to extraterrestrial irradiance
+              on a horizontal plane.
+
+    References
+    -------
+    .. [1] Louche A, Notton G, Poggi P, Simmonnot G. Correlations for direct
+    normal and global horizontal irradiation on French Mediterranean site.
+    Solar Energy
+    1991;46:261-6
+
+    """
+    bool = np.logical_or(solar_zenith > 180, solar_zenith < 0)
+    solar_zenith = np.where(bool, np.NaN, solar_zenith)
```

Comment:
In an abundance of caution, it may be better to store this result in a local variable rather than modifying `solar_zenith` in place.

Reply:
Any suggestions for internal variable of solar_zenith & datetime_or_doy? I thought solar_zenith_valid